tss/rsa/rsa_threshold_test.go

```diff
@@ -0,0 +1,419 @@
+	if err != nil {
+		t.Fatal(err)
+	}
+	testIntegration(t, algo, &pub, players, threshold, keys, PSS)
+}
+
+func benchmarkSignCombineHelper(randSource io.Reader, parallel bool, b *testing.B, players, threshold int64, bits int, algo crypto.Hash, padScheme int) {
+	key, err := rsa.GenerateKey(rand.Reader, bits)
+	pub := key.PublicKey
+	if err != nil {
+		panic(err)
+	}
+
+	keys, err := Deal(rand.Reader, players, threshold, key, true)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	msg := []byte("hello")
+	var padder Padder
+	if padScheme == PKS1v15 {
+		padder = &PKCS1v15Padder{}
+	} else if padScheme == PSS {
+		padder = &PSSPadder{
+			Rand: rand.Reader,
+			Opts: nil,
+		}
+	} else {
+		b.Fatal(errors.New("unknown padScheme"))
+	}
+	msgPH, err := PadHash(padder, algo, &pub, msg)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	signshares := make([]SignShare, threshold)
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		for i := int64(0); i < threshold; i++ {
+			signshares[i], err = keys[i].Sign(randSource, players, &pub, msgPH, parallel)
+			if err != nil {
+				b.Fatal(err)
+			}
+		}
+		_, err = CombineSignShares(players, &pub, signshares, msgPH)
+		if err != nil {
+			b.Fatal(err)
+		}
+	}
+	b.StopTimer()
+}
+
+func BenchmarkBaselineRSA_SHA256_4096(b *testing.B) {
+	const bits = 4096
+	const algo = crypto.SHA256
+
+	key, err := rsa.GenerateKey(rand.Reader, bits)
+	if err != nil {
+		b.Fatal(err)
+	}
+	h := algo.New()
+
+	msg := []byte("hello")
+
+	h.Write(msg)
+	d := h.Sum(nil)
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		_, err = rsa.SignPKCS1v15(rand.Reader, key, algo, d)
+		if err != nil {
+			b.Fatal(err)
+		}
+	}
+	b.StopTimer()
+}
+
+func BenchmarkBaselineRSA_SHA256_2048(b *testing.B) {
+	const bits = 2048
+	const algo = crypto.SHA256
+
+	key, err := rsa.GenerateKey(rand.Reader, bits)
+	if err != nil {
+		b.Fatal(err)
+	}
+	h := algo.New()
+
+	msg := []byte("hello")
+
+	h.Write(msg)
+	d := h.Sum(nil)
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		_, err = rsa.SignPKCS1v15(rand.Reader, key, algo, d)
+		if err != nil {
+			b.Fatal(err)
+		}
+	}
+	b.StopTimer()
+}
+
+func BenchmarkSignCombine_SHA256_4096_3_2_Scheme(b *testing.B) {
+	const players = 3
+	const threshold = 2
+	const bits = 4096
+	const algo = crypto.SHA256
+	benchmarkSignCombineHelper(nil, false, b, players, threshold, bits, algo, PKS1v15)
+}
+
+func BenchmarkSignCombine_SHA256_4096_3_2_Scheme_Blind(b *testing.B) {
+	const players = 3
+	const threshold = 2
+	const bits = 4096
+	const algo = crypto.SHA256
+	benchmarkSignCombineHelper(rand.Reader, false, b, players, threshold, bits, algo, PKS1v15)
+}
+
+func BenchmarkSignCombine_SHA256_4096_3_2_Scheme_BlindParallel(b *testing.B) {
+	const players = 3
+	const threshold = 2
+	const bits = 4096
+	const algo = crypto.SHA256
+	benchmarkSignCombineHelper(rand.Reader, true, b, players, threshold, bits, algo, PKS1v15)
+}
+
+func BenchmarkSignCombine_MD5_4096_3_2_Scheme(b *testing.B) {
```

Comment:
MD5 is no more in use.

Reply:
I think it's nice to see how much of the bench is hashing but we can remove

Reply:
yes let's remove it, usually hashing is not issue for performance.